src/store/auth/actions.js

```diff
@@ -1,52 +1,207 @@
+      });
   },
 
-  _logOut({ commit }) {
-    commit('logOut');
-    localStorage.removeItem('user');
-    localStorage.removeItem('currentListId');
-    deleteAccessToken();
-    commitFromRoot('sidebar/closeSidebar');
-    commitFromRoot('filters/resetFilters');
-    dispatchFromRoot('visualization/_resetVisualizationToDefault');
-    commitFromRoot('setCurrentListItems', []);
-    router.push({ name: 'auth' });
+  _resetPassword(state, { email, code, password }) {
+    commitFromRoot('increaseExplicitRequestsNumber');
+
+    return this.$config.axios.post(
+      `${this.$config.apiBasePath}auth/reset-password`,
+      { 
+        email,
+        code,
+        password,
+      },
+    )
+      .then(() => {
+        commitFromRoot('setNotification', { text: 'Password reset is successful' });
+        router.push({ name: 'signIn' });
+      })
+      .catch(error => {
+        throw getErrorMessage(error.response.data);
+      })
+      .finally(() => {
+        commitFromRoot('decreaseExplicitRequestsNumber');
+      });
+  },
+
+  _changePassword(state, {
+    currentPassword,
+    newPassword,
+    isLogoutFromAllDevices,
+  }) {
+    const fingerprint = JSON.stringify(getBrowserFingerprint());
+    const user = JSON.parse(localStorage.getItem('user'));
+
+    commitFromRoot('increaseExplicitRequestsNumber');
+
+    return this.$config.axios.post(
+      `${this.$config.apiBasePath}auth/change-password`,
+      { 
+        email: user.email,
+        accessToken: user.accessToken,
+        refreshToken: user.refreshToken,
+        currentPassword,
+        newPassword,
+        isLogoutFromAllDevices,
+        fingerprint,
+      },
+    )
+      .then(() => {
+        commitFromRoot('setNotification', { text: 'Password changed successfully' });
+      })
+      .catch(error => {
+        throw getErrorMessage(error.response.data);
+      })
+      .finally(() => {
+        commitFromRoot('decreaseExplicitRequestsNumber');
+      });
+  },
+
+  _logOut({ commit }, { mode }) {
+    const fingerprint = JSON.stringify(getBrowserFingerprint());
+    const user = JSON.parse(localStorage.getItem('user'));
+
+    commitFromRoot('increaseExplicitRequestsNumber');
+
+    return this.$config.axios.post(
+      `${this.$config.apiBasePath}auth/logout`,
+      {
+        mode,
+        fingerprint,
+        accessToken: user.accessToken,
+        refreshToken: user.refreshToken,
+        userId: user.id,
+      },
+    )
+      .then(() => {
+        if (mode !== 'allExceptCurrent') {
+          commit('logOut');
+          localStorage.removeItem('user');
+          localStorage.removeItem('currentListId');
+          deleteAccessToken();
+          commitFromRoot('sidebar/closeSidebar');
+          commitFromRoot('filters/resetFilters');
+          commitFromRoot('setCurrentListItems', []);
+          dispatchFromRoot('visualization/_resetVisualizationToDefault');
+          router.push({ name: 'auth' });
```

Comment:
I would like to propose you to add some empty lines between logical groups of actions here